scrape/manager.go

```diff
@@ -208,6 +211,39 @@ func (m *Manager) reload() {
 	}
 	m.mtxScrape.Unlock()
 	wg.Wait()
+
+	if m.opts.WarnDuplicateTargets {
+		m.warnIfTargetsRelabelledToSameLabels()
+	}
+}
+
+func (m *Manager) warnIfTargetsRelabelledToSameLabels() {
+	m.mtxScrape.Lock()
+	defer m.mtxScrape.Unlock()
+
+	totalTargets := 0
+	for _, scrapePool := range m.scrapePools {
+		totalTargets += len(scrapePool.activeTargets)
+	}
+
+	activeTargets := make(map[uint64]*Target, totalTargets)
+	builder := labels.NewBuilder(labels.EmptyLabels())
+	for _, scrapePool := range m.scrapePools {
+		for _, target := range scrapePool.activeTargets {
+			tHash := target.hash()
+			t, ok := activeTargets[tHash]
+			if !ok {
+				activeTargets[tHash] = target
+				continue
+			}
+			m.logger.Warn(
+				"Found targets with same labels after relabelling",
+				"target_one", t.DiscoveredLabels(builder).Get(model.AddressLabel),
```

Comment:
if the labels are the same, the address label value should be identical, unless there is a hash collision.
logging the labels is enough.

Also, in `docs/feature_flags.md` we may want to explicitly say that the log will contain the labels, in case someone is storing some sensitive data in there.